### ISSUE TITLE
[Settings] Hide (previously Adapt) General/Discovery settings

### ISSUE THREAD

**Issue by giomfo** [tchapgouv/tchap-android-v2#251]
Can you please check first why we get the following exception on this screen (this may be due to the missing terms from id server) 
![image](https://user-images.githubusercontent.com/8969772/138740110-0352f69c-7159-4e49-be91-70ff3cd6e28f.png)

-> **this exception is because the endpoint `_matrix/identity/v2/terms` is blocked server side currently**

- [x] Hide the first section "Serveur d'identité"
- [x] Update the second section by removing phone number references
![image](https://user-images.githubusercontent.com/8969772/138740430-a0d89a52-aa50-4538-a0cd-8986d84166e5.png)
- [x] Hide the last section, phone number are not available
![image](https://user-images.githubusercontent.com/8969772/138740535-f9fb65c6-3bc9-46bf-a573-87a8619ee6bb.png)

**Comment by giomfo** [comment]
I will mark this task as Done because the DINUM decided to hide for the moment the contact book access in Tchap
This section named "Discovery settings" will be hidden for the moment (we don't need to ask consent about id server to the user for the moment)

**Comment by giomfo** [comment]
I realized Element-Android requires this consent to trigger any lookup requests 
Without this consent, Element-Android doesn't reveal existing user while you fill a complete their email address in the new direct screen
![image](https://user-images.githubusercontent.com/8969772/142658197-67baad0a-2068-4b77-ae1c-bb2f0d8545c5.png)

In Tchap, I don't think we need this additional consent because the id server is forced by the Tchap infra. The consent to send email addresses to discover other Tchap users is granted when the user accept the global Tchap TAC (https://www.tchap.gouv.fr/tac/) , see the last section

@Anna-Livia what do you think? can we hide this additional consent request in Tchap?
if not we have to define tchap id server terms and open _matrix/identity/v2/terms endpoint on each id server

**Comment by Anna-Livia** [comment]
I'm having trouble understanding what the feature does.

What I understand the use case is: 
While using Element, if I want to start a conversation with someone whom I never had a conversation with, I can use their email address to look them up and start a chat room. 

Is that right ?

Then, in Tchap, once you do that, the app will show the display name of the user that have that email adresse and allow you to send them an invite.

However, and this is where I get confused, what happens in element android ?

**Comment by giomfo** [comment]
Element and Tchap are currently doing the same: they are waiting for user's consent before sending the email to the server and eventually help you to discover an existing user, this prevent you from inviting them by email. You will directly start a chat in Tchap.

My question here is should we keep the requirement about the user's consent before sending email to the server.
It was not the case in TchapV1 because we considered accepted TAC was enough.

The Element-Android UI provide us a way to prompt the user before sending email to the server. We may keep it if you think this is useful in Tchap

**Comment by jdauphant** [comment]
For me the user is ok to send the email currently typed.
(but not the contacts repository)

This will simplify the implementation?

**Comment by giomfo** [comment]
@jdauphant yes, we will just hide this section from the general settings, so we will not have to customise the screen.

I will unblock this issue by changing the demand
-[ ] hide section "Discovery settings" from General settings

@Florian14 can you please drop your changes suggested in #PR 267, and prepare a new PR for the above request

**Comment by jdauphant** [comment]
In Element Android FOSS, is that possible to auto-hide the menu if the matrix server return 403 ?

**Comment by giomfo** [comment]
@jdauphant no, this menu doesn't depend only on the terms of the id server.
We may discuss in direct about the discovery feature which is enabled by default in Tchap now
